core/src/main/java/org/elasticsearch/index/termvectors/TermVectorsService.java

```diff
@@ -103,12 +96,12 @@ static TermVectorsResponse getTermVectors(IndexShard indexShard, TermVectorsRequ
             Versions.DocIdAndVersion docIdAndVersion = get.docIdAndVersion();
             /* from an artificial document */
             if (request.doc() != null) {
-                termVectorsByField = generateTermVectorsFromDoc(indexShard, request, !docFromTranslog);
+                termVectorsByField = generateTermVectorsFromDoc(indexShard, request, true);
```

Comment:
this seems to be the only call site of `generateTermVectorsFromDoc` so maybe we can simplify `generateTermVectorsFromDoc` by removing all branches that handle the `false` case